api4/license.go

```diff
@@ -94,7 +97,27 @@ func addLicense(c *Context, w http.ResponseWriter, r *http.Request) {
 	buf := bytes.NewBuffer(nil)
 	io.Copy(buf, file)
 
-	license, appErr := c.App.Srv().SaveLicense(buf.Bytes())
+	licenseBytes := buf.Bytes()
+	license, appErr := utils.LicenseFromBytes(licenseBytes)
+	if appErr != nil {
+		c.Err = appErr
+		return
+	}
+
+	if license.IsTrial() {
+		can, err := c.App.Srv().LicenseManager.CanStartTrial()
+		if err != nil {
+			c.Err = model.NewAppError("addLicense", "api.license.add_license.open.app_error", nil, "", http.StatusInternalServerError)
+			return
+		}
+
+		if !can {
+			c.Err = model.NewAppError("addLicense", "api.license.request-trial.can-start-trial.not-allowed", nil, "", http.StatusBadRequest)
+			return
+		}
+	}
+
+	license, appErr = c.App.Srv().SaveLicense(licenseBytes)
```

Comment:
It won't block if the trial extension is more than (or actually even less than) 30 days as for now we're relying on the duration.

Once the `IsTrial` flag is merged and we start using it, it will be blocked. I think the extensions would have to be for more or less than 30 days even with the flag.